readthedocs/projects/constants.py

```diff
@@ -431,3 +431,23 @@
         _("Single version without translations (/<filename>)"),
     ),
 )
+
+
+ADDONS_FLYOUT_SORTING_LEXICOGRAPHYCALLY = "alphabetically"
+# Compatibility to keep the behavior of the old flyout.
+# This isn't a good algorithm, but it's a way to keep the old behavior in case we need it.
+ADDONS_FLYOUT_SORTING_SEMVER_READTHEDOCS_COMPATIBLE = "semver-readthedocs-compatible"
+# https://pypi.org/project/packaging/
+ADDONS_FLYOUT_SORTING_PYTHON_PACKAGING = "python-packaging"
+ADDONS_FLYOUT_SORTING_CALVER = "calver"
+# Let the user to define a custom pattern and use BumpVer to parse and sort the versions.
+# https://github.com/mbarkhau/bumpver#pattern-examples
+ADDONS_FLYOUT_SORTING_CUSTOM_PATTERN = "custom-pattern"
+
+ADDONS_FLYOUT_SORTING_CHOICES = (
+    (ADDONS_FLYOUT_SORTING_LEXICOGRAPHYCALLY, "Alphabetically"),
+    (ADDONS_FLYOUT_SORTING_SEMVER_READTHEDOCS_COMPATIBLE, "SemVer (Read the Docs)"),
+    (ADDONS_FLYOUT_SORTING_PYTHON_PACKAGING, "Python Packaging (PEP 440 and PEP 425)"),
+    (ADDONS_FLYOUT_SORTING_CALVER, "CalVer (YYYY.0M.0M)"),
+    (ADDONS_FLYOUT_SORTING_CUSTOM_PATTERN, "Define your own pattern"),
```

Comment:
He, we _need_ a differentiation from the original SemVer itself because our algorithm is a pretty customized version of SemVer. I kept it here because is the current algorithm we are using in the old flyout and will help with the migration, but we need to differentiate it from the _real SemVer_.

> I don't think users will known what `(Read the Docs)` implies here.

We need documentation for each of these algorithms.

Reply:
Yeah, so I wouldn't worry much about describing the options in the option text then. I would do that in documentation. I also feel more complex UI would really solve usability issues here too, but I'd want to see a fair amount of usage here first.